xrootd/client/file.go

```diff
@@ -55,6 +56,22 @@ func (f file) Sync(ctx context.Context) error {
 	return err
 }
 
+// ReadAtContext reads len(p) bytes into p starting at offset off.
+func (f file) ReadAtContext(ctx context.Context, p []byte, off int64) (n int, err error) {
+	var resp read.Response
```

Comment:
actually, couldn't we directly create the response pointing at the destination `p` ?
something like:
```go
resp := read.Response{Data:p}
```

Reply:
actually, it seems the `copy` is necessary (at least, the way we currently implement things).
at the very least, it should be:
```go
resp := read.Response{Data: make([]byte, 0, len(p))}
```